pandas/tests/io/test_orc.py

```diff
@@ -301,3 +303,45 @@ def test_orc_writer_dtypes_not_supported(df_not_supported):
     msg = "The dtype of one or more columns is not supported yet."
     with pytest.raises(NotImplementedError, match=msg):
         df_not_supported.to_orc()
+
+
+def test_orc_use_nullable_dtypes_pandas_backend_not_supported(dirpath):
+    input_file = os.path.join(dirpath, "TestOrcFile.emptyFile.orc")
+    with pytest.raises(
+        NotImplementedError,
+        match="io.nullable_backend set to pandas is not implemented.",
+    ):
+        with pd.option_context("io.nullable_backend", "pandas"):
+            read_orc(input_file, use_nullable_dtypes=True)
+
+
+@td.skip_if_no("pyarrow", min_version="7.0.0")
+def test_orc_use_nullable_dtypes_pyarrow_backend():
+    df = pd.DataFrame(
+        {
+            "string": list("abc"),
+            "string_with_nan": ["a", np.nan, "c"],
+            "string_with_none": ["a", None, "c"],
+            "bytes": [b"foo", b"bar", None],
+            "int": list(range(1, 4)),
+            "float": np.arange(4.0, 7.0, dtype="float64"),
+            "float_with_nan": [2.0, np.nan, 3.0],
+            "bool": [True, False, None],
+            "datetime": pd.date_range("20130101", periods=3),
+            "datetime_with_nat": [
+                pd.Timestamp("20130101"),
+                pd.NaT,
+                pd.Timestamp("20130103"),
+            ],
+        }
+    )
+    bytes_data = df.to_orc()
```

Comment:
Just to avoid something subtle: can you do df.copy().to… since you are using df below?

Reply:
Good idea. Added the copy